src/devices/DCMotorController/DCMotorController.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Pwm;
+using System.Device.Pwm.Drivers;
+
+namespace Iot.Device
+{
+    public class DCMotorController : IDisposable
+    {
+        private const double PwmFrequency = 50; // Hz
+        private PwmController _pwm;
+        private GpioController _controller;
+        private int _d0;
+        private int? _d1;
+        private double _speed;
+
+        /// <summary>
+        /// Initialize a DCMotorController class
+        /// </summary>
+        /// <param name="pin0">The GPIO pin number used for controlling the speed of the motor.</param>
+        /// <param name="pin1">The optional GPIO pin number used for controlling the direction of the motor.</param>
+        public DCMotorController(int pin0, int? pin1 = null)
+        {
+            _d0 = pin0;
+            _d1 = pin1;
+
+            _speed = 0;
+            _controller = new GpioController();
+            _pwm = new PwmController(new SoftPwm());
```

Comment:
Should we allow using hardware PWM as well?

Reply:
Feels wasteful considering this will usually work with around 50Hz. Is there any way to detect current pin is hardware PWM? Maybe we should have some kind of AutoPWM class which will auto-pick

Reply:
this should be addressed already